### ISSUE TITLE
[regression] next throws away my optional chaining and gives me a prerender error instead

### ISSUE THREAD

**Issue by yume-chan** [vercel/next.js#33915]
### Run `next info` (available from version 12.0.8 and up)

```
'yarn' is not recognized as an internal or external command,
operable program or batch file.
'pnpm' is not recognized as an internal or external command,
operable program or batch file.

    Operating System:
      Platform: win32
      Arch: x64
      Version: Windows 10 Pro
    Binaries:
      Node: 16.13.1
      npm: 8.1.2
      Yarn: N/A
      pnpm: N/A
    Relevant packages:
      next: 12.0.11-canary.1
      react: 17.0.2
      react-dom: 17.0.2
```

### What version of Next.js are you using?

12.0.11-canary.1

### What version of Node.js are you using?

v16.13.1

### What browser are you using?

Unrelated

### What operating system are you using?

Windows 11

### How are you deploying your application?

Unrelated

### Describe the Bug

With a super simple page like this

```js
import { value } from "../value"

export default function Home() {
  return (
    <div>
      {value?.not?.existed?.property}
    </div>
  )
}
```

and `value.js`

```js
export const value = {};
```

`next build` works with version 12.0.7, but not with 12.0.8 and later

```
info  - Checking validity of types
info  - Creating an optimized production build  
info  - Compiled successfully
info  - Collecting page data
[    ] info  - Generating static pages (0/3)
Error occurred prerendering page "/". Read more: https://nextjs.org/docs/messages/prerender-error
TypeError: Cannot read properties of undefined (reading 'existed')
    at Home (<project_folder>\.next\server\pages\index.js:29:29)
    at d (<project_folder>\node_modules\react-dom\cjs\react-dom-server.node.production.min.js:33:498)
    at bb (<project_folder>\node_modules\react-dom\cjs\react-dom-server.node.production.min.js:36:16)
    at a.b.render (<project_folder>\node_modules\react-dom\cjs\react-dom-server.node.production.min.js:42:43)
    at a.b.read (<project_folder>\node_modules\react-dom\cjs\react-dom-server.node.production.min.js:41:83)
    at Object.exports.renderToString (<project_folder>\node_modules\react-dom\cjs\react-dom-server.node.production.min.js:52:138)
    at Object.renderPage (<project_folder>\node_modules\next\dist\server\render.js:736:46)
    at Object.defaultGetInitialProps (<project_folder>\node_modules\next\dist\server\render.js:368:51)
    at Function.getInitialProps (<project_folder>\.next\server\pages\_document.js:530:20)
    at Object.loadGetInitialProps (<project_folder>\node_modules\next\dist\shared\lib\utils.js:69:29)
info  - Generating static pages (3/3)

> Build error occurred
Error: Export encountered errors on following paths:
        /
    at <project_folder>\node_modules\next\dist\export\index.js:499:19
    at runMicrotasks (<anonymous>)
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
    at async Span.traceAsyncFn (<project_folder>\node_modules\next\dist\trace\trace.js:75:20)
    at async <project_folder>\node_modules\next\dist\build\index.js:1014:17
    at async Span.traceAsyncFn (<project_folder>\node_modules\next\dist\trace\trace.js:75:20)
    at async <project_folder>\node_modules\next\dist\build\index.js:888:13
    at async Span.traceAsyncFn (<project_folder>\node_modules\next\dist\trace\trace.js:75:20)
    at async Object.build [as default] (<project_folder>\node_modules\next\dist\build\index.js:82:25)
```

---

`<project_folder>\.next\server\pages\index.js` produced by 12.0.7:

```js
function Home() {
    var ref, ref1;
    return(/*#__PURE__*/ jsx_runtime_namespaceObject.jsx("div", {
        children: value === null || value === void 0 ? void 0 : (ref = value.not) === null || ref === void 0 ? void 0 : (ref1 = ref.existed) === null || ref1 === void 0 ? void 0 : ref1.property
    }));
};
```

by 12.0.8

```js
function Home() {
    return(/*#__PURE__*/ jsx_runtime_namespaceObject.jsx("div", {
        children: value.not.existed.property
    }));
};
```

### Expected Behavior

It builds.

### To Reproduce

```
git clone https://github.com/yume-chan/next-ssg-regression.git
cd next-ssg-regression
npm i
npm run build
```

**Comment by kdy1** [comment]
I verified that it's already fixed on the main branch of swc.
Can you try it again with `v12.0.10`?

**Comment by yume-chan** [comment]
> I verified that it's already fixed on the main branch of swc.

Neither `v12.0.10` and `v12.0.11-canary.1` works. I updated my repo to include build cache.

Source: https://github.com/yume-chan/next-ssg-regression/blob/e9a0d3e1252414d317ed02936c44c82df0f78da1/pages/index.js#L3-L9

```js
export default function Home() {
  return (
    <div>
      {value?.not?.existed?.property}
    </div>
  )
}
```

Built: https://github.com/yume-chan/next-ssg-regression/blob/e9a0d3e1252414d317ed02936c44c82df0f78da1/.next/server/pages/index.js#L27-L31

```js
function Home() {
    return(/*#__PURE__*/ jsx_runtime_namespaceObject.jsx("div", {
        children: value.not.existed.property
    }));
};
```

**Comment by ijjk** [comment]
Hi, this has been updated in `v12.0.11-canary.5` of Next.js, please update and give it a try!

**Comment by yume-chan** [comment]
> this has been updated in `v12.0.11-canary.5`

Thanks! Verified it's fixed in `v12.0.11-canary.5`
